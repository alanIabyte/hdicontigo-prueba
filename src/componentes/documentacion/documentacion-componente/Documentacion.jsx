import React from "react";
import PropTypes from "prop-types";
import {
  DocumentacionCampo,
  DocumentacionValor,
  EnvolvedorDocumentacion,
  MensajeDocumentacion,
} from "./Documentacion.styled";

const diccionario = {
  estatus: "Estatus de la documentación:",
  mensaje: "Para ver más detalles del estatus de tu documentación visita",
  liga: "HDI Documenta",
};

const Documentacion = (props) => {
  const { datosDocumentacion } = props;

  const estatus = datosDocumentacion && datosDocumentacion.estatus;
  const liga = datosDocumentacion && datosDocumentacion.linkActual;

  return (
    <EnvolvedorDocumentacion>
      <DocumentacionCampo>{diccionario.estatus}</DocumentacionCampo>
      <DocumentacionValor>{estatus}</DocumentacionValor>
      <MensajeDocumentacion>
        {diccionario.mensaje}{" "}
        <a target="_blank" rel="noreferrer" href={liga}>
          {diccionario.liga}
        </a>
      </MensajeDocumentacion>
    </EnvolvedorDocumentacion>
  );
};

Documentacion.defaultProps = {
  datosDocumentacion: {},
};

Documentacion.propTypes = {
  datosDocumentacion: PropTypes.instanceOf(Object),
};

export default Documentacion;
