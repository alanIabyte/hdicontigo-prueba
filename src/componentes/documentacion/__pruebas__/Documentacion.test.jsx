import React from "react";
import renderer from "react-test-renderer";
import Documentacion from "../documentacion-componente/Documentacion";

describe("Documentacion", () => {
  it("se despliega correctamente", () => {
    const tree = renderer.create(<Documentacion />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
