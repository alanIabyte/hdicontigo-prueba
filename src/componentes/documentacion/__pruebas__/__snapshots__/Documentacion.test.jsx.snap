// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentacion se despliega correctamente 1`] = `
<div
  className="sc-ikkxIA"
>
  <div
    className="sc-kpDqfm sc-dAbbOL bTCkFB hdZOgq"
  >
    Estatus de la documentación:
  </div>
  <div
    className="sc-kpDqfm sc-feUZmu bTCkFB jMhLTa"
  />
  <div
    className="sc-iGgWBj sc-fUnMCh fgQoGq kBlNPQ"
  >
    Para ver más detalles del estatus de tu documentación visita
     
    <a
      rel="noreferrer"
      target="_blank"
    >
      HDI Documenta
    </a>
  </div>
</div>
`;
